core/sqf/src/seabed/src/Makefile

```diff
@@ -55,6 +55,16 @@ LIBSBUTIL	= $(LIBEXPDIR)/libsbutil.so
 LIBSQSTATESB	= $(LIBEXPDIR)/libsqstatesb.so
 # neet -lrt for clock_gettime
 LIBSX	       += -lrt
+#
+# the order of these objects have to be swapped based on the GCC version
+MS_MPITMSG_O	= $(OUTDIR)/mpitmsg.o \
+		  $(OUTDIR)/ms.o 
+ifeq ($(GCC_MAJOR_VERSION_GT_3),1)
+ifeq ($(GCC_MINOR_VERSION_GT_4),1)
```

Comment:
I'm not sure this is what you want. For example, if there is ever a gcc version 5.1, the first "ifeq" will be true but not the second.

Reply:
You are right Dave, so it should be GCC_MAJOR_VERSION_EQ_4 and GCC_MINOR_VERSION_GT_4 
We don't know how GCC 5.1 behave now, so let only check GCC 4.4 vs. GCC 4.8 at present. Let me change the logic more accurate
That is, we only change the link order for GCC 4.x , where x>4